modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -279,7 +282,24 @@ object Environment {
     if (preparedRemoteAdapters.nonEmpty)
       Clients
         .mkHttp(remoteAdapters.connectionTimeout, remoteAdapters.readTimeout, remoteAdapters.maxConnections, ec)
+        .map(enableRemoteAdapterMetrics(metrics))
         .map(c => (c.some, preparedRemoteAdapters))
     else Resource.pure[F, (Option[Http4sClient[F]], Map[(String, String), RemoteAdapter])]((None, preparedRemoteAdapters))
   }
+
+  def enableRemoteAdapterMetrics[F[_]: Clock: Sync](metrics: Metrics[F])(inner: Http4sClient[F]): Http4sClient[F] =
+    Http4sClient { req =>
+      for {
+        now <- Resource.eval(Clock[F].realTime(MILLISECONDS))
+        resp <- inner.run(req)
+        _ <- Resource.eval {
+               metrics.remoteAdaptersLatency(true, now) *>
+                 (resp.status.responseClass match {
+                   case Status.Successful => metrics.remoteAdapters2xxCount(true)
+                   case Status.ServerError => metrics.remoteAdapters5xxCount(true)
+                   case _ => Sync[F].unit
+                 })
```

Comment:
It would be great to capture timeouts, or failed to connect.  When you call `inner.run(req)` you only successfully get a response if the http call ends with a response.  But there is a whole different class of http error, where you don't get back a response at all!

I think you need something like `.handleErrorWith()` in order to capture the timeout exception.